From: Thomas Renninger <trenn@suse.de>
Subject: ACPI dock/bay: Use ACPI_EXCEPTION instead of printk(KERN_ERR
Patch-Mainline: not yet

Getting this error:
ACPI: \_SB_.PCI0.IDE1.PRI_.MAST: found ejectable bay
ACPI: \_SB_.PCI0.IDE1.PRI_.MAST: Adding notify handler
ACPI: Error installing bay notify handler

there could be various reasons why installing the notify handler failed.
It is expected that the dock driver already has installed the notify
handler for this device, but this is not clear.
printk suppresses this information.
The return value of the previous called and failed acpica function
holds this information and should be printed using ACPI_EXCEPTION which
is designed for printing error messages on failed acpica calls.

IMO this should get documented in Documentation/acpi/..., but I
was unsure whether it's worth to start a acpi_programming_rules.txt
(couldn't think of much more ading there for now) or
whether it's worth to add print_acpica_errors_in_kernel_drivers.txt
(or similar).

---
 drivers/acpi/bay.c  |    3 ++-
 drivers/acpi/dock.c |   10 ++++++----
 2 files changed, 8 insertions(+), 5 deletions(-)

--- a/drivers/acpi/bay.c
+++ b/drivers/acpi/bay.c
@@ -305,7 +305,8 @@ static int bay_add(acpi_handle handle, i
 	status = acpi_install_notify_handler(handle, ACPI_SYSTEM_NOTIFY,
 			bay_notify, new_bay);
 	if (ACPI_FAILURE(status)) {
-		printk(KERN_INFO PREFIX "Error installing bay notify handler\n");
+		ACPI_EXCEPTION((AE_INFO, status,
+				"Error installing bay notify handler\n"));
 		platform_device_unregister(new_bay->pdev);
 		goto bay_add_err;
 	}
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -415,8 +415,8 @@ static void handle_dock(struct dock_stat
 	arg.integer.value = dock;
 	status = acpi_evaluate_object(ds->handle, "_DCK", &arg_list, &buffer);
 	if (ACPI_FAILURE(status))
-		printk(KERN_ERR PREFIX "%s - failed to execute _DCK\n",
-			 (char *)name_buffer.pointer);
+		ACPI_EXCEPTION((AE_INFO, status, "%s - failed to execute"
+				" _DCK\n", (char *)name_buffer.pointer));
 	kfree(buffer.pointer);
 	kfree(name_buffer.pointer);
 }
@@ -834,7 +834,8 @@ static int dock_add(acpi_handle handle)
 					     dock_notify, dock_station);
 
 	if (ACPI_FAILURE(status)) {
-		printk(KERN_ERR PREFIX "Error installing notify handler\n");
+		ACPI_EXCEPTION((AE_INFO, status, "Error installing notify"
+				" handler\n"));
 		ret = -ENODEV;
 		goto dock_add_err;
 	}
@@ -877,7 +878,8 @@ static int dock_remove(void)
 					    ACPI_SYSTEM_NOTIFY,
 					    dock_notify);
 	if (ACPI_FAILURE(status))
-		printk(KERN_ERR "Error removing notify handler\n");
+		ACPI_EXCEPTION((AE_INFO, status,
+				"Error removing notify handler\n"));
 
 	/* cleanup sysfs */
 	device_remove_file(&dock_device->dev, &dev_attr_docked);
